ccw.core/src/clj/ccw/leiningen/generic_launch.clj

```diff
@@ -26,16 +26,16 @@
         args (str/trim args)]
     (launch/lein project-name args)))
 
-(defn generic-launch 
+(defn generic-launch
   "Open a popup asking the user the Leiningen command to issue,
    and then dynamically create a Java launch configuration invoking the command"
   [project]
-  (swt/ui display
-    (let [dialog	 (doto (swt/new-shell display
-                                       SWT/ON_TOP
-                                       ;SWT/TITLE
-                                       ;SWT/CLOSE
-                                       SWT/APPLICATION_MODAL)
+  (swt/ui (let [display (swt/display)
```

Comment:
I think that if we want to be able to correctly handle multi-display (multi-screens), which becomes more and more usual, we may have to not rely too often, implicitly, on (swt/display) which will return *some* display. What about forcing the user to be explicit like in

```clojure
(swt/ui [display workbench] ;; this resolves to a binding of display to (swt/display workbench) ;; swt/display being a Protocol
  ... use display (or d or whatever the binding has 
```

Not totally sure, will need some time to think about it